### PR TITLE
Utility to get document conflicts

### PR DESCRIPTION
**Pull request by hari-newredo**

### CODE REVIEW

File: bin/conflicts.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+var PouchDB = require('pouchdb');
```
Comment (pluby) [inline bin/conflicts.js]: Add "use strict".

File: bin/conflicts.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+PouchDB.plugin(require('pouchdb-find'));
+var fs = require('fs');
```
Comment (pluby) [inline bin/conflicts.js]: Use `const` instead of `var` wherever possible and use `let` otherwise.

File: bin/conflicts.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+var path = process.argv.slice(2);
+var db = new PouchDB(path[0]);
```
Comment (pluby) [inline bin/conflicts.js]: Should this be `...path = process.argv[2]...` then just `...new PouchDB(path)...`? Not sure please test.

File: bin/conflicts.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        "selector": {
+            "_conflicts": { "$exists": true}
+        }, 
+        "conflicts": true, 
+        "fields": [ "_id", "_rev", "_conflicts" ], 
+        "limit": 100 
+    });
+    for(let doc_id of conflicts.docs) {
+        const doc = await db.get(doc_id._id);
+        filename = doc_id._id + ' rev:' + doc_id._rev + ' primary' + '.txt';
```
Comment (pluby) [inline bin/conflicts.js]: `.json` instead of `.txt`?

File: bin/conflicts.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        fs.writeFileSync(filename, JSON.stringify(doc, null, 4), function(err) {
+            if(err) return console.error(err);
+        });
+        
+        for (let conflict of doc_id._conflicts) {
+            console.log(conflict);
+
+            cfilename = doc_id._id + ' rev:' + doc_id._rev + ' conflict' + conflict + '.txt';
```
Comment (pluby) [inline bin/conflicts.js]: `.json` instead of `.txt`?

File: bin/conflicts.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+            const conf = await db.get(doc_id._id, {rev:conflict});
+            fs.writeFileSync(cfilename, JSON.stringify(conf, null, 4), function(err) {
+                if(err) return console.error(err);
+            });
```
Comment (pluby) [inline bin/conflicts.js]: `writeFileSync` does not use callbacks, it throws errors (which there is no need to catch as they'll get caught higher up anyway and logged).

**Comment by pluby** [comment]
I get the error:

```
Error: Cannot find module 'pouchdb-find'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/tmp/src/bin/conflicts.js:12:16)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
```

Have you not checked in your `package.json` and `package-lock.json`?

**Comment by hari-newredo** [comment]
> I get the error:
> 
> ```
> Error: Cannot find module 'pouchdb-find'
>     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
>     at Function.Module._load (internal/modules/cjs/loader.js:562:25)
>     at Module.require (internal/modules/cjs/loader.js:692:17)
>     at require (internal/modules/cjs/helpers.js:25:18)
>     at Object.<anonymous> (/tmp/src/bin/conflicts.js:12:16)
>     at Module._compile (internal/modules/cjs/loader.js:778:30)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
>     at Module.load (internal/modules/cjs/loader.js:653:32)
>     at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
>     at Function.Module._load (internal/modules/cjs/loader.js:585:3)
> ```
> 
> Have you not checked in your `package.json` and `package-lock.json`?

pouchdb-find 7.2.2 is present in my package.json

**Comment by archanabattini** [comment]
Closed as its 3 years old

**Pull request closed without merging**